swagger.yaml

```diff
@@ -499,14 +540,209 @@ paths:
               schema:
                 $ref: "#/components/schemas/Error"
 
+  /v1/devices/{deviceId}/surveys:
+    get:
+      operationId: listDeviceSurveys
+      summary: Lists Device Surveys
+      description: Lists Device Surveys.
+      tags:
+        - DeviceSurveys
+      parameters:
+        - name: deviceId
+          required: true
+          in: path
+          description: Device id
+          schema:
+            type: string
+            format: uuid
+        - name: firstResult
+          in: query
+          description: First result. Defaults to 0
+          schema:
+            type: integer
+        - name: maxResults
+          in: query
+          description: Max results. Defaults to 10
+          schema:
+            type: integer
+      responses:
+        "200":
+          description: List of Device Surveys
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: "#/components/schemas/DeviceSurvey"
+          headers:
+            X-Total-Count:
+              schema:
+                type: integer
+                format: int64
+              description: Total count of Device Surveys with given filters.
+        "default":
+          description: Invalid request was sent to the server
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/Error"
+    post:
+      operationId: createDeviceSurvey
+      summary: Create a Device Survey.
+      description: Creates a new DeviceSurvey.
+      tags:
+        - DeviceSurveys
+      parameters:
+        - name: deviceId
+          required: true
+          in: path
+          description: Device id
+          schema:
+            type: string
+            format: uuid
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: "#/components/schemas/DeviceSurvey"
+        description: Payload
+        required: true
+      responses:
+        "200":
+          description: Created DeviceSurvey
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/DeviceSurvey"
+        "default":
+          description: Invalid request was sent to the server
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/Error"
+
+  /v1/devices/{deviceId}/surveys/{deviceSurveyId}:
+    get:
+      operationId: findDeviceSurvey
+      summary: Find a Device Survey.
+      description: Finds a Device Survey by id. Devices access this endpoint.
+      tags:
+        - DeviceSurveys
+      security:
+        - deviceAuth: []
```

Comment:
This is specified only for one endpoint. Is it supposed to be like that?

Reply:
Atleast for the time being this is the only endpoint accessed by devices.